packages/java/engine-runtime/src/test/java/com/vaadin/hilla/internal/AbstractTaskEndpointGeneratorTest.java

```diff
@@ -35,16 +42,67 @@ void shouldThrowIfOutputDirectoryIsNull() {
         }, "Output directory cannot be null");
     }
 
-    static private class TestTaskEndpointGenerator
+    private static final Function<String, URL> resourceFinder = Thread
```

Comment:
Is it needed to have this field static? Could potentially cause issues in parallel executions. What about providing the value directly in the call to `super(...)`?